x-pack/legacy/plugins/apm/public/utils/formatters/duration.ts

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import moment from 'moment';
+import { memoize } from 'lodash';
+import { NOT_AVAILABLE_LABEL } from '../../../common/i18n';
+import { asDecimal, asInteger } from './formatters';
+import { TimeUnit } from './datetime';
+import { Maybe } from '../../../typings/common';
+
+const SPACE = ' ';
+
+interface FormatterOptions {
+  withUnit?: boolean;
+  defaultValue?: string;
+}
+
+interface DurationTimeUnit {
+  [timeUnit: string]: {
+    translation: string;
```

Comment:
This should just be `label`. The interface doesn't need to know that it's a translated string underneath.

Reply:
sorry, not `label`. `unit` is probably better